server/schedule/filter/filters.go

```diff
@@ -461,9 +464,18 @@ func (f *ruleLeaderFitFilter) Source(opt opt.Options, store *core.StoreInfo) boo
 }
 
 func (f *ruleLeaderFitFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
-	oldLeaderPeerReplica := *f.region.GetLeader()
-	oldLeaderPeerReplica.StoreId = store.GetID()
-	region := f.region.Clone(core.WithReplacePeerStore(f.oldLeaderStoreID, store.GetID()), core.WithLeader(&oldLeaderPeerReplica))
+	var targetPeer *metapb.Peer
+	for _, peer := range f.region.GetPeers() {
+		if peer.StoreId == store.GetID() {
+			targetPeer = peer
+			break
+		}
+	}
+	if targetPeer == nil {
+		log.Info("ruleLeaderFitFilter could't find peer on target Store", zap.Uint64("newStoreID", store.GetID()))
```

Comment:
Will the log be noisy?

Reply:
I think normally it should not happen. Maybe `log.Warn` here.

Reply:
updated.